acra/configuring-maintaining/general-configuration/acra_keymaker.md

```diff
@@ -0,0 +1,178 @@
+  If destination storage is empty and hasn't any key then this argument is required. Read more about [keystore versions]({{< ref "/acra/keys/versions/" >}})
+  
+### Storage destination
+
+#### Filesystem
+
+* `--keys_output_dir=<folder>`
+  
+  Folder where will be saved keys (you can change destination for asymmetric public keys with keystore V1 (V2 store at same destination)  with parameter `--keys_public_output_dir`).  
+  Default is `.acrakeys`.
+
+* `--keys_public_output_dir=<folder>`
+  
+  Folder where will be saved public asymmetric keys. Works only for `--keystore=v1` and do nothing with `--keystore=v2`.
+  Default is `.acrakeys`.
+  
+#### Redis
+
+* `--redis_db_keys=`
+
+  Number of Redis database for keys.
+  Default is `0`.
+  <!-- `acra-server -help` says default is `-1` but in `cmd/redis.go` I see `redisDefaultDB = 0` -->
+  <!-- this var is also used as default value for the flag, where's the truth? -->
+
+* `--redis_host_port=<host:port>`
+
+  `<host:port>` used to connect to Redis.
+  Default is empty (don't connect).
+
+* `--redis_password=<password>`
+
+  Password to Redis database.
+
+
+### Configuration files
+
+* `--config_file=<filename>`
+
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-translator.yaml`.
+
+* `--generate_markdown_args_table`
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-translator.md`.
+  Works in pair with `--dump_config`.
+
+### Vault
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `http://x.x.x.x:yyyy`) for loading `ACRA_MASTER_KEY` from HashiCorp Vault.
+  Default is empty (`ACRA_MASTER_KEY` environment variable is expected).
+
+* `--vault_secrets_path=<path>`
+
+  KV Secret Path for reading `ACRA_MASTER_KEY` from HashiCorp Vault.
+  Default is `secret/`.
+
+* `--vault_tls_ca_path=<filename>`
+
+  Path to CA certificate for HashiCorp Vault certificate validation.
+  Default is empty (use root certificates configured in system).
+
+* `--vault_tls_client_cert=<filename>`
+
+  Path to client TLS certificate for reading `ACRA_MASTER_KEY` from HashiCorp Vault.
+  Default is empty (don't send client certificate).
+
+* `--vault_tls_client_key=<filename>`
+
+  Path to private key of the client TLS certificate for reading `ACRA_MASTER_KEY` from HashiCorp Vault.
+  Default is empty (don't send client certificate).
+
+* `--vault_tls_transport_enable={true|false}`
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+  
+### Common keys
+
+{{< hint warning >}}
+Note: backup all your keys before calling `acra-keymaker` with any flag described below. Some of them rotate and save current keys, 
+some of them may override current keys.
+{{< /hint >}}
+
+* `--generate_master_key`
+  
+  Create new random master key acceptable by all Acra tools and save to file.
```

Comment:
You need to tell which file will be used. It's also a nice place to remind how this file is supposed to be used.

```suggestion
* `--generate_master_key=<path>`

  Generate a new random master key and write it to the specified file.

  All Acra components will expect this key in `ACRA_MASTER_KEY` environment variable.
```

(Tricky question: how do you tell `acra-keymaker` to put the key into Vault instead?)

Reply:
nice question. created task for it) because it's good feature to allow generate and save master key directly to vault.